### ISSUE TITLE
OnMessage errors

### ISSUE THREAD

**Issue by Banaanae** [mmikeww/AHK-v2-script-converter#136]
AHK v1 code:
(modified example of OnMessage example no.1)
```ahk
Gui, Add, Text,, Click anywhere in this window.
Gui, Add, Edit, w200 vMyEdit
Gui, Show
OnMessage(0x0201, "WM_LBUTTONDOWN")
return

WM_LBUTTONDOWN(wParam, lParam)
{
    X := lParam & 0xFFFF
    Y := lParam >> 16
    if A_GuiControl
        Ctrl := "`n(in control " . A_GuiControl . ")"
    ToolTip You left-clicked in Gui window #%A_Gui% at client coordinates %X%x%Y%.%Ctrl%
}

F5::
Tooltip
OnMessage(0x0201, "")
Return

GuiClose:
ExitApp
```

Expected v2 code:
```ahk
MyGui := Gui()
MyGui.Add("Text",, "Click anywhere in this window.")
MyGui.Add("Edit", "w200")
MyGui.Show
OnMessage 0x0201, WM_LBUTTONDOWN

WM_LBUTTONDOWN(wParam, lParam, msg, hwnd)
{
    X := lParam & 0xFFFF
    Y := lParam >> 16
    Control := ""
    thisGui := GuiFromHwnd(hwnd)
    thisGuiControl := GuiCtrlFromHwnd(hwnd)
    if thisGuiControl
    {
        thisGui := thisGuiControl.Gui
        Control := "`n(in control " . thisGuiControl.ClassNN . ")"
    }
    ToolTip "You left-clicked in Gui window '" thisGui.Title "' at client coordinates " X "x" Y "." Control
}

F5:: {
Tooltip
OnMessage(0x0201, WM_LBUTTONDOWN, 0)
}
```

Generated Code:
```ahk
myGui := Gui()
myGui.OnEvent("Close", GuiClose)
myGui.Add("Text", , "Click anywhere in this window.")
ogcMyEdit := myGui.Add("Edit", "w200 vMyEdit")
myGui.Show()
OnMessage(0x0201, WM_LBUTTONDOWN)
return

WM_LBUTTONDOWN(wParam, lParam) ; also needs msg, hwnd
{
    X := lParam & 0xFFFF
    Y := lParam >> 16
    if A_GuiControl ; 
        Ctrl := "`n(in control " . A_GuiControl . ")"
    ToolTip("You left-clicked in Gui window #" A_Gui " at client coordinates " X "x" Y "." Ctrl)
}

F5::
{ ; V1toV2: Added bracket
ToolTip()
OnMessage(0x0201) ; Should be OnMessage(0x0201, WM_LBUTTONDOWN, 0)
Return
} ; Added bracket before function

GuiClose(*)
{ ; V1toV2: Added bracket
ExitApp()

} ; Added bracket in the end
```

**Comment by Banaanae** [comment]
Failing tests added in #137

**Comment by Banaanae** [comment]
Currently:

OnMessage now turns off only if its define before its turned off
One OnMessage param (does nothing) acts as turning off
Multiple OnMessage with same MsgNumber (rare) requires more testing

**Comment by andymbody** [comment]
@Banaanae ... I'm having trouble understanding this issue. Why is  (wParam, lParam, msg, hwnd) expected? What is the criteria for adding msg and hwnd?

**Comment by andymbody** [comment]
@Banaanae ... clarification... is all that is needed here is to make sure the params are/become
(wParam, lParam, msg, hwnd) ?

**Comment by Banaanae** [comment]
Sorry I missed your comment, that was the main thing, but the other issue was OnMessage being converted if it was turned off before it was turned on, there is a fix in #184, I just need to get around to taking it out of there and into here
